MiscVBAFunctions/Modules/Test__MiscDataContainers.bas

```diff
@@ -0,0 +1,237 @@
+Attribute VB_Name = "Test__MiscDataContainers"
+Option Explicit
+Option Private Module
+
+'@TestModule
+'@Folder("Tests")
+
+Private Assert As Rubberduck.AssertClass
+Private Fakes As Rubberduck.FakesProvider
+
+'@ModuleInitialize
+Private Sub ModuleInitialize()
+    'this method runs once per module.
+    Set Assert = New Rubberduck.AssertClass
+    Set Fakes = New Rubberduck.FakesProvider
+End Sub
+
+'@ModuleCleanup
+Private Sub ModuleCleanup()
+    'this method runs once per module.
+    Set Assert = Nothing
+    Set Fakes = Nothing
+End Sub
+
+'@TestInitialize
+Private Sub TestInitialize()
+    'This method runs before every test in the module..
+End Sub
+
+'@TestCleanup
+Private Sub TestCleanup()
+    'this method runs after every test in the module.
+End Sub
+
+'@TestMethod("MiscDataStructures")
+Private Sub Test_Join_Collections()
+    On Error GoTo TestFail
+
+    'Arrange:
+    Dim z, x, y As New Collection
```

Comment:
I don't think this sets z and x as Collections? It should be `Dim z as Collection, x as Collection, y as Collection`. If we had the `JoinCollections()` function that would pick the type mismatches up